### PR TITLE
Outfactor Image

### PR DESCRIPTION
**Pull request by disko**
**This is still work in progress.  Please do not merge, but review and / or coment.**

Prepare outfactoring of the Image content type into a separate package:

  - add a Depot tween and file serve app based on and providing the same features as Depot's WSGI middleware.
  - refactor SaveDataMixin, File and Image to support the use of different filter sets on file content types (for example image thumbnails).

Rationale
---------

Depot supports the concepts of attachments and filters.  These can be used for example to create thumbnails for images on creation time (currently image resizing in Kotti is done on the fly during a request).  Other use cases could be text extraction for PDF or office file types, video conversion, etc.

This PR implements the infrastructure to ease implementation of these use cases.

It is planned to outfactor Image into a separate ``kotti_image`` package, as not all applications need the specialized behavior of this content type (the more generic File content type is often enough).  By outfactoring the Image behavior could be made configurable in much greater detail without adding a bunch of configuration options to the Kotti core.  We also would get rid of the ``plone.scale`` and ``Pillow`` dependencies.  In Kotti 1.x we would import Image from ``kotti_image`` to keep the current behavior, but mark it as deprecated.  Kotti 2 would remove that import and make ``kotti_image`` completely optional.

### CODE REVIEW

File: kotti/resources.py [hunk 1]
```diff
@@ -655,6 +660,27 @@ class SaveDataMixin(object):
...
+    #: (:class:`sqlalchemy.types.String`)
+    mimetype = Column(String(100))
+    #: Size of the file in bytes
+    #: (:class:`sqlalchemy.types.Integer`)
+    size = Column(Integer())
+
+    #: Filedepot mapped blob
+    #: (:class:`depot.fileds.sqlalchemy.UploadedFileField`)
+    @declared_attr
+    def data(cls):
```
Comment (disko) [inline kotti/resources.py]: Codacy Issue found: [An attribute defined in kotti.resources line 753 hides this method](https://www.codacy.com/app/disko/Kotti/file/1578056223/issues/source?bid=2089747&fileBranchId=3023493#l677)

**Comment by dnouri** [comment]
Sounds like a great plan to me.

**Comment by disko** [comment]
FWIW: ready to merge.

If nobody complains, I'll merge and release as 1.3.0-alpha.1 tomorrow.  Would feel better if someone could at least do a quick review though.  TIA!

File: kotti/resources.py [hunk 2]
```diff
@@ -655,6 +655,27 @@ class SaveDataMixin(object):
...
```
Comment (disko) [inline kotti/resources.py]: Codacy Issue found: [An attribute defined in kotti.resources line 753 hides this method](https://www.codacy.com/app/disko/Kotti/file/1591525434/issues/source?bid=2089747&fileBranchId=3023493#l672)

Comment (disko) [inline kotti/resources.py]: Codacy Issue found: [An attribute defined in kotti.resources line 753 hides this method](https://www.codacy.com/app/disko/Kotti/file/1597238718/issues/source?bid=2089747&fileBranchId=3023493#l672)

Comment (disko) [inline kotti/resources.py]: Codacy Issue found: [An attribute defined in kotti.resources line 753 hides this method](https://www.codacy.com/app/disko/Kotti/file/1602142554/issues/source?bid=2089747&fileBranchId=3023493#l672)

File: kotti/resources.py [hunk 3]
```diff
@@ -655,6 +654,27 @@ class SaveDataMixin(object):
...
```
Comment (disko) [inline kotti/resources.py]: Codacy Issue found: [An attribute defined in kotti.resources line 753 hides this method](https://www.codacy.com/app/disko/Kotti/file/1604698433/issues/source?bid=2089747&fileBranchId=3023493#l671)

Comment (disko) [inline kotti/resources.py]: Codacy Issue found: [An attribute defined in kotti.resources line 753 hides this method](https://www.codacy.com/app/disko/Kotti/file/1634933313/issues/source?bid=2089747&fileBranchId=3023493#l671)

**Comment by disko** [comment]
Implemented / merged in 9f7d4a9ff6b0a93ac4aaa68bfe7d04fa166d98c6

**Pull request closed without merging**